### PR TITLE
Use base realm to serve host app assets

### PR DESCRIPTION
**Pull request by habdelra**
TODO:
- [x] setup script to establish sym links in dev env
- [x] update deploy to push host dist into the base realm in hosted env
- [x] test thru bare origin realm
- [x] host realm-server's dom-test container
- [x] update README.md to reflect the different build types and how to start the app and how to run the tests 

**NOTE: you will mostly likely need to remove your old service worker first after pulling this into your workspace if you are running into problems running your app** 

This PR uses the base realm to host the ember assets (and test assets used for the realm server's DOM test which tests the realm's ability to serve an ember app). The key here is that we actually update the webpack build to add a path prefix to the assets that correlates to where the assets are now  served on the base realm. This is important because not only does the index.html need to be updated, the dynamically loaded chunk.js files that webpack adds logic to request properly need to be loaded from the same place. This results in an index.html that is incompatible with an ember-cli hosted app. 

This PR includes the ability to build 2 kinds of apps: an app that can be hosted by ember cli--useful for development, and an app that can be hosted by the realm server where the assets come from the base realm (this can also be used for development too). The normal ember-cli hosted app is run in the same way, via `pnpm start` in the host package. For the app that uses base realm to serve it's assets, start the an ember build watch using `pnpm start:build` and then start your realm server(s) (which should include the base realm). Navigate to the realm server in your browser to view the host app for the specific realm server--in this mode the host assets will come from the base realm.

Also, a new capability fell out of this approach, which is actually pretty cool that allows a server to serve both a hosted realm and a local realm from the same server. This capability is enabled via a flag on the server start script `--hostLocalRealm`. When this flag is enabled, the `http://server-name/local` becomes available which serves the local realm. We have enabled this flag on the realm server that serves the base realm in development, such that you can visit `http://localhost:4201/local` to view the local realm. As well as `http://localhost:4201/base` will show the base realm's app. (might be kinda fun to turn this on in staging)

Additionally, to run the realm server's DOM tests, navigate to `http://localhost:4201/base/__tests` (on the base realm). This will run the realm servers DOM tests in QUnit.

As part of deployment, we perform 2 types of builds: a build for the ember app hosted by s3/cloudfront and a build for the ember app hosted by the realm server,  so that you can visit either the current staging env (s3/cloudfront hosted app) or you can visit a realm server directly in your browser. We might want to discuss retiring the ember app that is hosted by s3/cloudfront (although we probably still want to use this infra to host the dist.zip that the realm server consumes).

As part of this in dev mode, when you start the base realm server (either individually or via `start:all`) symlinks are automatically created so that the base realm reflects the current build (as well as any realm server DOM test updates). And in the deploy scripts we deploy the ember build to the base realm as part of the deployment process. This uses the same zip file approach to download the zip from s3/cloudfront, although we no longer use the fastboot zip. The fastboot zip does not respect the ember-cli-build config that we performed for webpack to alter the asset path. As such we build a dist for the realm server hosting, zip it up, and push to s3 in a separate step during the host deploy.

Finally, another change that this introduces is to no longer deploy to the server that hosts only the base realm. We were actually never using this server (we might in the future) so this will speed up deployments by just not deploying to that server (that server is still running, though, if we change our minds in the future).

### CODE REVIEW

File: packages/host/app/config/environment.d.ts [hunk 1]
```diff
@@ -13,9 +13,10 @@ declare const config: {
   rootURL: string;
   APP: Record<string, unknown>;
   ownRealmURL: string;
-  localRealmEnabled: boolean;
+  isLocalRealm: boolean;
   resolvedBaseRealmURL: string;
-  servedByRealm: boolean;
+  isBaseRealmHosting: boolean;
+  realmsServed?: string[];
```
Comment (habdelra) [inline packages/host/app/config/environment.d.ts:19]: take a look at `packages/worker/src/fetch.ts` for why this was added

File: packages/host/app/services/local-realm.ts [hunk 2]
```diff
@@ -321,6 +323,7 @@ export default class LocalRealm extends Service {
         send(registration.installing!, {
           type: 'setDirectoryHandle',
           handle: this.state.handle,
+          realmsServed,
```
Comment (habdelra) [inline packages/host/app/services/local-realm.ts:326]: Take a look at `packages/worker/src/fetch.ts` for why this was added

**Review by lukemelia — APPROVED** [review]
Looks good. I ran it locally and everything seems to work. Once the README is updated and TODOs are captured, good by me.

**Comment by habdelra** [comment]
I'm a little conflicted about hosting the test assets in the base realm. we only ever use this for dev and CI--in staging and prod it would actually require us hosting a test realm in order to work. As part of supporting this we had to add custom logic into the realm and server modules to support the test routes. seems overkill for something that is not used in prod. Id suggest going back to running an http server just for these tests and getting rid of the special test logic in our server and realm modules.

**Comment by habdelra** [comment]
Set the publicAsssetURL at runtime using global variable instead of at build time with env var

**Comment by habdelra** [comment]
it would be better if we didn't need to use a zip for the dist. and instead all assets are served via http

**Comment by habdelra** [comment]
It looks like the runtime global to set the public asset path, `__webpack_public_path__` doesn't seem to work. We should follow up separately on this
image

**Pull request merged**